src/electionguard-ui/ElectionGuard.UI/ViewModels/ManifestViewModel.cs

```diff
@@ -0,0 +1,130 @@
+﻿using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+
+namespace ElectionGuard.UI.ViewModels;
+
+public record PartyDisplay(string Name, string Abbreviation, string PartyId);
+public record CandidateDisplay(string Name, string Party, string CandidateId);
+public record BallotStyleDisplay(string Name, List<string> GeoPoliticalUnits);
+public record GeopoliticalUnitDisplay(string Name, string GPType, string GeopoliticalUnitId);
+public record ContestDisplay(string Name, string Variation, ulong NumberElected, ulong VotesAllowed, List<CandidateDisplay> Selections);
+
+
+public partial class ManifestViewModel : BaseViewModel
+{
+    private ManifestService _manifestService;
+
+    public ManifestViewModel(IServiceProvider serviceProvider, ManifestService manifestService) : base("ManifestViewer", serviceProvider)
+    {
+        _manifestService = manifestService;
+    }
+
+    [ObservableProperty]
+    private Manifest? _manifest;
+
+    [ObservableProperty]
+    private string _manifestName;
+
+    [ObservableProperty]
+    private string _manifestFile = string.Empty;
+
+    [ObservableProperty]
+    private ObservableCollection<PartyDisplay> _parties = new();
+
+    [ObservableProperty]
+    private ObservableCollection<CandidateDisplay> _candidates = new();
+
+    [ObservableProperty]
+    private ObservableCollection<BallotStyleDisplay> _ballotStyles = new();
+
+    [ObservableProperty]
+    private ObservableCollection<GeopoliticalUnitDisplay> _geopoliticalUnits = new();
+
+    [ObservableProperty]
+    private ObservableCollection<ContestDisplay> _contests = new();
+
+
+    partial void OnManifestFileChanged(string value)
+    {
+        if (ManifestFile == null || ManifestFile == string.Empty)

```

Comment:
We should use `String.IsNullOrEmpty()` here

Reply:
fixed